src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ImportStatusBauService.cs

```diff
@@ -46,29 +45,32 @@ public class ImportStatusBauService : IImportStatusBauService
                     var queryResults = await _tableStorageService.ExecuteQueryAsync(
                         DatafileImportsTableName, tableQuery);
 
-                    return queryResults.Where(result =>
+                    return queryResults.Select(result =>
                     {
-                        if (result.Message == "")
-                        {
-                            _logger.LogWarning(
-                                $"{MethodBase.GetCurrentMethod()} received Imports row with empty Message. " +
-                                $"PartitionKey: \"{result.PartitionKey}\", RowKey: \"{result.RowKey}\"");
-                            return false;
-                        }
+                        var releaseId = new Guid(result.PartitionKey);
+                        var dataFileName = result.RowKey;
+                        var release =
+                            _contentDbContext.Releases
+                                .Where(r => r.Id == releaseId)
+                                .Include(r => r.Publication)
+                                .First();
 
-                        return true;
-                    }).Select(result =>
-                    {
-                        var m = JsonConvert.DeserializeObject<ImportMessage>(result.Message);
+                        var releaseFileRef = _contentDbContext.ReleaseFileReferences.First(r =>
+                            r.ReleaseId == releaseId && 
+                            r.Filename == dataFileName &&
+                            r.ReleaseFileType == ReleaseFileTypes.Data);
+
+                        var subjectId = releaseFileRef.SubjectId ?? Guid.Empty;
```

Comment:
I think I would just let the SubjectId be null below, rather than using Guid.Empty